lib/native/http.ts

```diff
@@ -291,6 +291,9 @@ export class HttpClientConnectionManager extends NativeResource {
                         handle
                     );
                     this.connections.set(handle, connection as HttpClientConnection);
+                    connection.on('close', () => {
```

Comment:
Bug 1: connections lived forever, even when they were dead, causing them to leak